grakn-graph/src/main/java/ai/grakn/factory/AbstractInternalFactory.java

```diff
@@ -91,18 +91,21 @@ public synchronized M getGraph(boolean batchLoading){
         if(batchLoading){
             batchLoadingGraknGraph = getGraph(batchLoadingGraknGraph, true);
             lastGraphBuiltBatchLoading = true;
+            batchLoadingGraknGraph.openTransaction();
             return batchLoadingGraknGraph;
         } else {
             graknGraph = getGraph(graknGraph, false);
             lastGraphBuiltBatchLoading = false;
+            graknGraph.openTransaction();
             return graknGraph;
         }
     }
     protected M getGraph(M graknGraph, boolean batchLoading){
+        boolean hasCommitted = false;
         //This checks if the previous graph built with this factory is the same as the one we trying to build now.
+        //TODO: Check is this features is still needed. It's very confusing at the end of the day
```

Comment:
Basically this weird thing is to force a refresh when someone commits with the batch graph. If we don't force a refresh then the normal graph is out of date. We originally added this because analytics was constantly switching between the batch and normal graph. The question is, do we still need it? @duckofyork in your opinion?